rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerLegacyImplTest.java

```diff
@@ -0,0 +1,397 @@
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.peg.utils;
+
+import co.rsk.bitcoinj.core.*;
+import co.rsk.config.BridgeConstants;
+import co.rsk.config.BridgeRegTestConstants;
+import co.rsk.core.RskAddress;
+import co.rsk.crypto.Keccak256;
+import co.rsk.peg.*;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ConsensusRule;
+import org.ethereum.core.Block;
+import org.ethereum.core.Transaction;
+import org.ethereum.util.RLP;
+import org.ethereum.util.RLPElement;
+import org.ethereum.util.RLPList;
+import org.ethereum.vm.DataWord;
+import org.ethereum.vm.LogInfo;
+import org.ethereum.vm.PrecompiledContracts;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test class for BridgeEventLoggerLegacyImpl.
+ *
+ * @author kelvin.isievwore
+ */
+
+public class BridgeEventLoggerLegacyImplTest {
+
+    private static final BridgeRegTestConstants CONSTANTS = BridgeRegTestConstants.getInstance();
+    private ActivationConfig.ForBlock activations;
+    private List<LogInfo> eventLogs;
+    private BrigeEventLoggerLegacyImpl eventLogger;
+    private BridgeConstants constantsMock;
+    private BtcTransaction btcTxMock;
+    private BtcTransaction btcTx;
+    private Keccak256 rskTxHash;
+
+    @Before
+    public void setup() {
+        activations = mock(ActivationConfig.ForBlock.class);
+        eventLogs = new LinkedList<>();
+        constantsMock = mock(BridgeConstants.class);
+        eventLogger = new BrigeEventLoggerLegacyImpl(constantsMock, activations, eventLogs);
+        btcTxMock = mock(BtcTransaction.class);
+        btcTx = new BtcTransaction(CONSTANTS.getBtcParams());
+        rskTxHash = PegTestUtils.createHash3(1);
+    }
+
+    @Test
+    public void testLogUpdateCollectionsBeforeRskip146() {
+        when(activations.isActive(ConsensusRule.RSKIP146)).thenReturn(false);
```

Comment:
You cold move this line to the `setup` function so it's not repeated in every test

Reply:
It Has To Be Repeated Depending On The Method, We Could Have `.thenReturn(false)` Or `.thenReturn(true)`

Reply:
Oh I see, thanks. So when RSKIP146 is active we simply expect a DeprecatedMethodCallException right? In that case I suggest we simplify those tests since there is no need to pass real values.

Something like this should do it
```
eventLogger.logReleaseBtc(any(), any());
````

Reply:
Resolved